src/Entities.js

```diff
@@ -17,6 +17,31 @@ limitations under the License.
 var React = require('react');
 var sdk = require('./index');
 
+function isMatch(query, name, uid) {
+    query = query.toLowerCase();
+    name = name.toLowerCase();
+    uid = uid.toLowerCase();
+
+    // direct prefix matches
+    if (name.indexOf(query) === 0 || uid.indexOf(query) === 0) {
+        return true;
+    }
+
+    // strip @ on uid and try matching again
+    if (uid.length > 1 && uid.substring(1).indexOf(query) === 0) {
+        return true;
+    }
+
+    // split spaces in name and try matching constituient parts
+    var parts = name.split(" ");
+    for (var i = 0; i < parts.length; i++) {
+        if (parts[i].indexOf(query) === 0) {
```

Comment:
As per IRL, it doesn't work great if you do this because this is the *display name* and disambig on display names with user IDs ain't gonna work (for one, we put the UID in brackets so `[0]` won't be `@` it will be `(`).